### PR TITLE
Pause/Resume other music when read aloud is enabled

### LINKED ISSUE #535: Stop music from other apps when Read Aloud is enabled.
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request

### Step 1: Are you ready to report?
   - Your issue may already be reported! Please search on the [issue track](../) before creating one.
   - [x] I have tested with the last (dev) version of the software.  ( If not, and if possible, run your test again with the most recent version available.)

### Step 2: What is your environment?
  * version of the software you use : 2.3
  * device / operating system : OnePlus 5T / Android Version 8.0
  
### Step 3: Describe the problem:

  * What do you obtain? A copy of the error message or a screenshot is always useful.
When read aloud is enabled, the sound overlaps with any other music playing from other apps.
  * When does this occur? 

#### Steps to reproduce:

  1. Play music from any other application.
  2. Select Read aloud from the menu.
  
  * What should be the behaviour from your point of view? How do you expect the service to work?
The music from other applications should be paused when read aloud is enabled.

### PR DESCRIPTION
**Pull request by Abishek10**
Fixes #535 

Changes: Use audio manager to listen to audio focus changes and play / pause text-to-speech based on that.

Screenshots/GIF for the change: NA

### CODE REVIEW

**Comment by Abishek10** [comment]
@brijeshshah13 @mhutti1 Please review this PR.

File: app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java [hunk 1]
```diff
@@ -617,7 +618,7 @@ public void onSpeakingStarted() {
         isSpeaking = true;
         runOnUiThread(() -> {
           menu.findItem(R.id.menu_read_aloud)
-              .setTitle(createMenuItem(getResources().getString(R.string.menu_read_aloud_stop)));
+                  .setTitle(createMenuItem(getResources().getString(R.string.menu_read_aloud_stop)));
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java]: Please revert unnecessary changes.

**Comment by Abishek10** [comment]
@srv-twry Done 👍

File: app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java [hunk 2]
```diff
@@ -632,7 +633,25 @@ public void onSpeakingEnded() {
           pauseTTSButton.setText(R.string.tts_pause);
         });
       }
-    });
+    }, new AudioManager.OnAudioFocusChangeListener() {
+      @Override
+      public void onAudioFocusChange(int focusChange) {
+        Log.d("Focus has changed", String.valueOf(focusChange));
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java]: Please look how we use Log.d in other places.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java [hunk 3]
```diff
@@ -35,6 +37,12 @@
 
   public TTSTask currentTTSTask = null;
 
+  private AudioManager audioManager;
+
+  final Object mFocusLock;
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java]: Don't use Hungarian anymore please.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java [hunk 4]
```diff
@@ -109,9 +120,19 @@ public void readAloud(WebView webView) {
       } else {
         tts.setLanguage(locale);
 
-        // We use JavaScript to get the content of the page conveniently, earlier making some
-        // changes in the page
-        webView.loadUrl("javascript:" +
+        if(requestAudioFocus() == true) {
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java]: Google Java style guide. spacing. Please check this as it is an issue in multiple locations.

**Comment by Abishek10** [comment]
@mhutti1 Done.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java [hunk 5]
```diff
@@ -135,14 +152,31 @@ public void stop() {
...
+      }
+    }
+  }
+
   public void pauseOrResume() {
     if (currentTTSTask == null)
       return;
 
-    if (currentTTSTask.paused)
+    if (currentTTSTask.paused) {
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/KiwixTextToSpeech.java]: Could we convert this to an if/else if. This is really messy. Remember spacing: } else {
